hypha/apply/stream_forms/models.py

```diff
@@ -74,8 +77,8 @@ def get_form_fields(self):
 
         return form_fields
 
-    def get_form_class(self):
-        return type('WagtailStreamForm', (self.submission_form_class,), self.get_form_fields())
+    def get_form_class(self, draft=False):
+        return type('WagtailStreamForm', (self.submission_form_class,), self.get_form_fields(draft))
```

Comment:
@sks444 Ideally `stream_forms` app should be considered like a third party app and funds related changes should happen in the apply app. So, we should try to not make changes here. Instead, we can make changes in [SubmittableStreamForm](https://github.com/OpenTechFund/hypha/blob/a0f9dcaf889e92f535a76dd082e5e816a3bd0af3/hypha/apply/funds/models/utils.py#L58) which is inherited by [RoundBase](https://github.com/OpenTechFund/hypha/blob/a0f9dcaf889e92f535a76dd082e5e816a3bd0af3/hypha/apply/funds/models/applications.py#L145). By overriding the `get_form_class` method in it.

Reply:
Looks like we will have to copy all the code in `get_form_fields` method if we override it in SubmittableStreamForm class. So, we can send the list of blocks `(EmailBlock, FullNameBlock, TitleBlock)` which should not have required False along with the draft value which calling `get_form_class(draft)` method. Instead of hard coding the block names here.

Reply:
Yeah we might have to copy all the code which doesn't look good.

Idea of sending the list of blocks as tuple sounds good. But it feels a bit weird to send all the those block names to get_form_class, that condition will be dependent on the draft value of it anyway. I was thinking if there is way to only check for the block from which it is inherited `ApplicationMustIncludeFieldBlock` and not do required False for them.

Reply:
Good idea. Checking `ApplicationMustIncludeFieldBlock` is better than option than checking individual blocks like `EmailBlock`.